scipy/interpolate/_bsplines.py

```diff
@@ -561,6 +601,71 @@ def integrate(self, a, b, extrapolate=None):
         integral *= sign
         return integral.reshape(ca.shape[1:])
 
+    @classmethod
+    def from_cubic(cls, cb, axis=0, bc_type='natural', extrapolate=None):
```

Comment:
OK, let's think about the API. First of all, this function _in general_ can accept any piecewise polynomial, it's only the continuity constraint that forces you to only accept CubicSpline objects for the time being. (if it were only CubicSplines, the implementation could have been much simpler just using explicit formulas, and here you're keeping it open for further extensions.)

Then maybe name it `from_ppform` (matlab/ deBoor brevity) or `from_power_basis` or some such (specific name I'm open to suggestions) and in the very beginning of the function have something like

```
if not isinstance(cb, CubicSpline):
    raive NotImplementedError("Only CubicSpline objects are accepted for now.")
```

BTW, my other suggestion to move it all to `_bspl.pyx` gets trickier: keep the public-facing function with the docstring and the check in python, and delegate heavy lifting to a private cython function.

Further on, why having `bc_type`, `axis` and `extrapolate` arguments if we're only accepting a CubicSpline instance --- just read them off from it.

Reply:
I've removed ``axis`` and ``extrapolate`` arguments in the new version of the function but it still needs to take ``bc_type`` because ``CubicSpline`` does not have it in attributes